packages/contractkit/src/wrappers/SortedOracles.ts

```diff
@@ -24,4 +122,61 @@ export class SortedOraclesWrapper extends BaseWrapper<SortedOracles> {
       reportExpirySeconds: await this.reportExpirySeconds(),
     }
   }
+
+  /**
+   * Helper function to get the rates for StableToken, by passing the address
+   * of StableToken to `getRates`.
+   */
+  getStableTokenRates = async (): Promise<OracleRate[]> => this.getRates(CeloContract.StableToken)
+
+  /**
+   * Gets all elements from the doubly linked list.
+   * @param token The CeloToken representing the token for which the Celo
+   *   Gold exchange rate is being reported. Example: CeloContract.StableToken
+   * @return An unpacked list of elements from largest to smallest.
+   */
+  async getRates(token: CeloToken): Promise<OracleRate[]> {
+    const tokenAddress = await this.kit.registry.addressFor(token)
+    const response = await this.contract.methods.getRates(tokenAddress).call()
+    const rates: OracleRate[] = []
+    for (let i = 0; i < response[0].length; i++) {
+      const medRelIndex = parseInt(response[2][i], 10)
+      rates.push({
+        address: response[0][i],
+        rate: new BigNumber(response[1][i]),
+        medianRelation: medRelIndex,
+      })
+    }
+    return rates
+  }
+
+  private async findLesserAndGreaterKeys(
+    token: CeloToken,
+    numerator: number,
+    denominator: number
+  ): Promise<{ lesserKey: Address; greaterKey: Address }> {
+    const currentRates: OracleRate[] = await this.getRates(token)
+    const internalDenominator = new BigNumber(await this.contract.methods.DENOMINATOR().call())
+
+    // This is how the contract calculates the rate from the numerator and denominator.
+    // To figure out where this new report goes in the list, we need to compare this
+    // value with the other rates
+    const value = internalDenominator.times(numerator).div(denominator)
+    let greaterKey = NULL_ADDRESS
+
+    // This leverages the fact that the currentRates are already sorted from
+    // greatest to lowest value
+    for (const rate of currentRates) {
+      if (!eqAddress(rate.address, this.kit.defaultAccount)) {
```

Comment:
instead of using kit defaultaccount, expect an argument. This pattern has been wrongly used in other wrapper and we are in the middle of changing that.
The reason is simple, we can't really assume that the `from` will be the defaultAccount, since it's just a default. So `report()` should expect the oracleAddress.

Also, `toTransactionObject()` now accepts a third argument on which you can specify a different default from; check `Validators.reorderMember()` for an example

Reply:
Got it. Should all functions that send a transaction take a `from` argument?

Reply:
No, actually wrapper functions don't really send the tx, but return a TransactionObject that the user can then use to send the tx (and they can specify a from there).

The problem arises when you need the "from" before; to do some pre-calculation, as in this case where you need that to compute lesser/greater keys. In those case, then ask for the `from` within the method parameters